Index: Tools/scons-addons/src/SConsAddons/Options/Options.py
===================================================================
--- Tools/scons-addons/src/SConsAddons/Options/Options.py	(revision 283)
+++ Tools/scons-addons/src/SConsAddons/Options/Options.py	(working copy)
@@ -48,10 +48,13 @@
 
 __revision__ = "__FILE__ __REVISION__ __DATE__ __DEVELOPER__"
 
-import types, string, os.path
+import types, string, os.path,sys
 import SCons.Errors
 import SCons.Util
-import textwrap
+if sys.version[:3] == "2.2":
+    import optik.textwrap as textwrap
+else:
+    import textwrap
 pj = os.path.join
 
 import SCons.SConf
