--- opennurbs_array_defs.h	2016-09-25 17:08:10.813869319 +0200
+++ patched/opennurbs_array_defs.h	2016-09-25 17:10:41.134082537 +0200
@@ -53,6 +53,17 @@
 //  Class ON_SimpleArray<>
 /////////////////////////////////////////////////////////////////////////////////////
 
+ON_DECL
+void ON_qsort(void*, size_t, size_t, int (*)(void*, const void*, const void*), void*);
+
+ON_DECL
+void ON_qsort( 
+        void* base,
+        size_t count,
+        size_t sizeof_element,
+        int (*compare)(const void*,const void*)
+        );
+
 // construction ////////////////////////////////////////////////////////
 
 template <class T>
@@ -1459,7 +1470,7 @@
 #else
   // for lamer 64 bit compilers
   return found ? ((int)((((ON__UINT64)found) - ((ON__UINT64)m_a))/sizeof(T))) : -1;
-#endif#endif
+#endif
 }
 
 template <class T>
@@ -1476,7 +1487,7 @@
 #else
   // for lamer 64 bit compilers
   return found ? ((int)((((ON__UINT64)found) - ((ON__UINT64)m_a))/sizeof(T))) : -1;
-#endif#endif
+#endif
 }
 
 template <class T>
