--- opennurbs_array.h	2011-11-22 09:51:36.000000000 +0800
+++ patched/opennurbs_array.h	2012-08-31 11:36:48.000000000 +0800
@@ -1719,7 +1719,7 @@
 See Also:
   ON_CompareDecreasing
 */
-template< class T>
+template< class T> static
 int ON_CompareIncreasing( const T* a, const T* b);
 
 /* 
@@ -1752,7 +1752,7 @@
 See Also:
   ON_CompareIncreasing
 */
-template< class T>
+template< class T> static
 int ON_CompareDecreasing( const T* a, const T* b);
 
 
