commit fa092012d67e9a44b6a71cade9d06adf7a5a1c5a
Author: gerrit <gerrit@feynman.camtech.ntu.edu.sg>
Date:   Sun Jan 2 15:28:47 2011 +0800

    fixed: never ending recursion

diff --git a/src/util.cpp b/src/util.cpp
index 76af586..a2a9c85 100644
--- a/src/util.cpp
+++ b/src/util.cpp
@@ -3231,7 +3231,7 @@ static QCString getCanonicalTypeForIdentifier(
     //result = mType->typeString();
     if (word!=mType->typeString())
     {
-      result = getCanonicalTypeForIdentifier(d,fs,mType->typeString(),tSpec,count++);
+      result = getCanonicalTypeForIdentifier(d,fs,mType->typeString(),tSpec,++count);
     }
     else
     {
